gradio/helpers.py

```diff
@@ -893,3 +896,50 @@ def __init__(self, target: Block | None, _data: Any):
         """
         self.target = target
         self._data = _data
+
+
+def log_message(message: str, level: Literal["info", "warning"] = "info"):
+    from gradio import context
+
+    if not hasattr(context.thread_data, "blocks"):  # Function called outside of Gradio
+        if level == "info":
+            print(message)
+        elif level == "warning":
+            warnings.warn(message)
+        return
+    if not context.thread_data.blocks.enable_queue:
+        warnings.warn(
+            f"Queueing must be enabled to issue {level.capitalize()}: '{message}'."
+        )
+        return
+    context.thread_data.blocks._queue.log_message(
+        event_id=context.thread_data.event_id, log=message, level=level
+    )
+
+
+@document()
+class Warning:
+    """
+    This class allows you to pass custom warning messages to the user. You can do so simply with `gr.Warning('message here')`, and when that line is executed the custom message will appear in a modal on the demo.
```

Comment:
```suggestion
    This class allows you to pass custom warning messages to the user. You can do so simply by instantiating `gr.Warning('message here')`, and when that line is executed the custom message will appear in a modal on the demo.
```

Reply:
actually changing this to a function, no need to be a class.